### PR TITLE
Issue#17 multimedia dashboard

### PR DESCRIPTION
**Pull request by mattporritt**
Update the report summary to add an initial chart to display file breakdown

### CODE REVIEW

File: classes/output/report_summary.php [hunk 1]
```diff
@@ -126,6 +126,70 @@ private function calculate_total_cost() {
...
+        if (!empty(($values))) { // Handle case where there is no data in table
+            $totalfiles = $values['totalfiles']->value;
+            $videofiles = $values['videofiles']->value;
+            $audiofiles = $values['audiofiles']->value;
+
+            $otherfiles = $totalfiles - ($videofiles + $audiofiles);
+            $totals = array($otherfiles, $videofiles, $audiofiles);
+        }
+
+        return $totals;
```
Comment (tomdickman) [inline classes/output/report_summary.php]: Minor thing, but won't this error when there are no files, as `$total` is never initialised?

File: classes/task/report_process.php [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * A scheduled task.
```
Comment (tomdickman) [inline classes/task/report_process.php]: Maybe make this more descriptive?

File: classes/output/report_summary.php [hunk 3]
```diff
@@ -126,6 +126,70 @@ private function calculate_total_cost() {
...
+    }
+
+    /**
+     * Generate the markeup for the file sumamry chart,
```
Comment (tomdickman) [inline classes/output/report_summary.php]: spelling 'summary'

File: classes/task/report_process.php [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+            $transaction->rollback($e);
+        }
+    }
+
+    /**
+     * Do the job.
+     * Throw exceptions on errors (the job will be retried).
+     */
+    public function execute() {
+        global $DB;
```
Comment (tomdickman) [inline classes/task/report_process.php]: `$DB` not used

File: classes/output/report_summary.php [hunk 5]
```diff
@@ -126,6 +126,70 @@ private function calculate_total_cost() {
...
+        $videofiles = 0;
+        $audiofiles = 0;
+        $otherfiles = 0;
+
+        // Get values for chart from the database.
+        list($insql, $inparams) = $DB->get_in_or_equal(array('totalfiles', 'videofiles', 'audiofiles'));
+        $select = "name $insql";  // Don't count files added by smartmedia itself.
+        $values = $DB->get_records_select('local_smartmedia_reports', $select, $inparams, '', 'name, value');
+
+        if (!empty(($values))) { // Handle case where there is no data in table
```
Comment (tomdickman) [inline classes/output/report_summary.php]: Fullstop on comment, don't know why Travis didn't pick it up.

File: tests/report_process_test.php [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        $name = 'totalfiles';
+        $value = 64;
+
+        $task = new \local_smartmedia\task\report_process();
+
+        // We're testing a private method, so we need to setup reflector magic.
+        $method = new ReflectionMethod('\local_smartmedia\task\report_process', 'update_report_data');
+        $method->setAccessible(true); // Allow accessing of private method.
+        $proxy = $method->invoke($task, $name, $value); // Get result of invoked method.
+        $proxy = $method->invoke($task, $name, $value); // Get result of invoked method.
```
Comment (tomdickman) [inline tests/report_process_test.php]: Do we need to invoke the method twice for the test?

File: classes/output/report_summary.php (hunk 1 above)
Comment (tomdickman) [reply classes/output/report_summary.php]: And by error I mean, return `null`.

File: classes/output/report_summary.php (hunk 3 above)
Comment (tomdickman) [reply classes/output/report_summary.php]: And 'markep'

File: classes/output/report_summary.php (hunk 1 above)
Comment (mattporritt) [reply classes/output/report_summary.php]: $totals is initialised as an empty array on line 137 as a guard against this condition

File: tests/report_process_test.php (hunk 6 above)
Comment (mattporritt) [reply tests/report_process_test.php]: yeps, first time invokes insert, second time invokes update in the upsert method being tested.  
calling it twice make sure both cases are handled and an exception isn't thrown

**Comment by mattporritt** [comment]
Changes made based on review

**Pull request merged**